src/components/Drawer/__tests__/Drawer.spec.js

```diff
@@ -2,50 +2,128 @@ import React from 'react';
 import Drawer from '../Drawer';
 
+jest.mock('react-transition-group', () => {
+    const FakeTransition = jest.fn(({ children }) => children({ state: 'entered' }));
+    return { Transition: FakeTransition };
+});
+
 describe('Drawer', () => {
     it('should render correctly', () => {
-        const wrapper = shallow(<Drawer title="some title">some text</Drawer>);
+        const wrapper = mount(<Drawer title="some title">some text</Drawer>);
         expect(toJson(wrapper)).toMatchSnapshot();
     });
 
-    it('should pass initial open status', () => {
-        const wrapper = shallow(
-            <Drawer initialIsOpen title="some title">
+    it('should pass initial expand status', () => {
+        const wrapper = mount(
+            <Drawer initialIsExpand title="some title">
                 some text
             </Drawer>
         );
 
-        expect(wrapper.find('.Drawer').hasClass('Drawer--isOpen')).toBe(true);
-        expect(wrapper.find('.Drawer__extendButton').hasClass('Drawer__extendButton--isOpen')).toBe(
-            true
-        );
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndExpanded')).toBe(true);
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndClosed')).toBe(false);
+        expect(
+            wrapper.find('.Drawer__expandButton').hasClass('Drawer__expandButton--isExpanded')
+        ).toBe(true);
     });
 
     it('should pass initial close status', () => {
-        const wrapper = shallow(<Drawer title="some title">some text</Drawer>);
+        const wrapper = mount(<Drawer title="some title">some text</Drawer>);
+
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndClosed')).toBe(true);
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndExpanded')).toBe(false);
+        expect(
+            wrapper.find('.Drawer__expandButton').hasClass('Drawer__expandButton--isExpanded')
+        ).toBe(false);
+    });
+
+    it('should expand and close correctly', () => {
+        const wrapper = mount(<Drawer title="some title">some text</Drawer>);
+        // Expand Drawer
+        wrapper.find('.Drawer__expandButton').simulate('click');
+
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndExpanded')).toBe(true);
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndClosed')).toBe(false);
+        expect(
+            wrapper.find('.Drawer__expandButton').hasClass('Drawer__expandButton--isExpanded')
+        ).toBe(true);
+
+        // Close Drawer
+        wrapper.find('.Drawer__expandButton').simulate('click');
+
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndClosed')).toBe(true);
+        expect(wrapper.find('.Drawer').hasClass('Drawer--isShownAndExpanded')).toBe(false);
+        expect(
+            wrapper.find('.Drawer__expandButton').hasClass('Drawer__expandButton--isExpanded')
+        ).toBe(false);
+    });
+
+    it('should expand and close correctly', () => {
```

Comment:
is this test the same as the previous one?

Reply:
Thanks!